Add post navigation links to handbooks

ryelle opened the pull request:
See #161 — This adds the post navigation links to the bottom of handbooks. It needs to use the handbook hierarchy for determining which is next/previous, so I had to filter those links to use the correct post. The links should now reflect the next or previous item in the Chapter List sidebar.

On the production site, this navigates through the items slightly differently — the previous item will take you to the previous section at the same level you're on, not the previous item in the list. For example, the previous link on [Design Contributions](https://developer.wordpress.org/block-editor/contributors/design/) is "Code Contributions", but on this PR it's "How To Get Your Pull Request Reviewed". I can't get that working right now, but I might just need fresh eyes.

| block-editor | coding-standards | plugins | rest-api | 
|----|----|----|---|
| ![](https://user-images.githubusercontent.com/541093/216472980-537cf1a7-f0c5-48ae-8bba-01d0d9e243b0.png) | ![](https://user-images.githubusercontent.com/541093/216472985-c63bd14e-8aae-43a6-8da9-11ee5650c103.png) | ![](https://user-images.githubusercontent.com/541093/216472989-d7a34592-1afd-450d-a916-9456a1d37471.png) | ![](https://user-images.githubusercontent.com/541093/216472991-9aab33e1-2a2b-4148-888e-63e3f3df7345.png) |

**To test**

- Navigate through a few different handbook pages
- The next/prev buttons should work
- The path they take you should match the Chapter List order
- The first item should have no "previous", and the last item should have no "Next"

ryelle committed:
Add post nav links

ryelle committed:
Update the navigation links using the same order as the chapter list

ryelle committed:
Clean up unused vars, only run the filter on handbooks

StevenDufresne commented:
On: http://localhost:8888/block-editor/getting-started/devenv/,

The next page is [ColorPalette](http://localhost:8888/block-editor/reference-guides/components/color-palette/). That is unexpected. Is there something wrong with my environment?

StevenDufresne commented on source/wp-content/themes/wporg-developer-2023/functions.php:
```diff
@@ -297,8 +299,8 @@ function pre_get_posts( $query ) {
 function register_nav_menus() {
```
I think this whole function is dead now.

ryelle replied:
Probably, but that's actually not related to this change:

- `devhub-cli-menu` is the nav in the blue banner on [the WP-CLI handbook](https://developer.wordpress.org/cli/commands/)
- `devhub-menu` is the nav in the blue banner everywhere else, but it's empty
- `reference-home-api` is the API column on [the reference page](https://developer.wordpress.org/reference/)

The 3 handbooks that don't use github do use nav menus on production for their ToC — I mentioned that in https://github.com/WordPress/wporg-developer/pull/189 but didn't implement that split behavior. 

On production, they're just unassigned menus used in a widget.
<img width="464" alt="Screenshot 2023-02-03 at 1 17 26 PM" src="https://user-images.githubusercontent.com/541093/216677549-b493e10e-dd9a-49fc-a5e3-870ccb2c6f4a.png">

All that is to say, I'd rather not remove this as part of this PR, but we can remove it in general.

ryelle committed:
Fix logic error so filter runs on the handbooks

ryelle commented:
> The next page is [ColorPalette](http://localhost:8888/block-editor/reference-guides/components/color-palette/). That is unexpected. Is there something wrong with my environment?

No… 🤦🏻  I introduced a logic error in https://github.com/WordPress/wporg-developer/pull/197/commits/5858762f32476e649912c479b33ae5a074a48b33 — that should be `if ( ! wporg_is_handbook() ) …`. Fixed in https://github.com/WordPress/wporg-developer/pull/197/commits/3a43effe6cc4f2d39bddfec528ed29cb7d11f675.

StevenDufresne reviewed (approved):
Yep, looks good.
